api/v1alpha1/releaseplan_types.go

```diff
@@ -17,20 +17,34 @@ limitations under the License.
 package v1alpha1
 
 import (
+	shared "github.com/redhat-appstudio/shared-utils/tekton"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
 )
 
 // ReleasePlanSpec defines the desired state of ReleasePlan.
 type ReleasePlanSpec struct {
-	// DisplayName is the long name of the ReleasePlan
-	// +optional
-	DisplayName string `json:"displayName"`
-
 	// Application is a reference to the application to be released in the managed namespace
 	// +kubebuilder:validation:Pattern=^[a-z0-9]([-a-z0-9]*[a-z0-9])?$
 	// +required
 	Application string `json:"application"`
 
+	// Data is an unstructured key used for providing data for the release Pipeline
+	// +kubebuilder:pruning:PreserveUnknownFields
+	// +optional
+	Data *runtime.RawExtension `json:"data,omitempty"`
+
+	// PipelineRef is an optional reference to a Pipeline that would be executed
```

Comment:
I think that it's fine here as we talk about a pipeline to be executed before the release pipeline... Later on, when it's finally implemented, we will have to add a name for sure... In my code I used `dev` and `managed` instead of `release`. But that code is reverted and will come in a separate PR.